DependencyInjection/DoctrineExtension.php

```diff
@@ -394,18 +397,36 @@ protected function ormLoad(array $config, ContainerBuilder $container)
             $container->removeAlias('doctrine.orm.metadata.annotation_reader');
         }
 
+        // available in Symfony 6.3
+        if (! class_exists(DoctrineDbalCacheAdapterSchemaListener::class)) {
+            $container->removeDefinition('doctrine.orm.listeners.doctrine_dbal_cache_adapter_schema_listener');
+        }
+
+        if (class_exists(DoctrineDbalCacheAdapterSchemaListener::class)) {
```

Comment:
it's not the same definition: see suffix, listener in one case, subscriber in the other

Reply:
I'm confused, I don't see suffix difference

<img width="705" alt="image" src="https://user-images.githubusercontent.com/496233/224496349-7b197869-203c-4085-9944-f4bf032670ef.png">

Reply:
ah, it's in the service name. The double "if" could be replaced by an if/else if that's more clear. It looks correct to me: if the class exists, remove A, else remove B, so that only one of (listener, subscriber) is registered.

Reply:
Since we are always removing a definition, let's replace with inline condition to determine service ID